### PR TITLE
smtube: use Qt5 on newer systems; depend on MPlayer via default variant, support QMPlay2 optionally

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

1. Use Qt5 on newer systems. Leave older on Qt4.
2. Do not force a user to build MPlayer if it is not desired. Instead, depend on it by default via a variant.
3. Support QMPlay2 via a non-default variant.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1
Xcode 15.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
I can confirm that with `+qmplay2` YouTube works on PowerPC. Previews of videos are still broken, but playback is fine:
image

File: www/smtube/Portfile [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 PortSystem          1.0
-PortGroup           qmake 1.0
+
+if {${os.platform} eq "darwin" && ${os.major} < 17} {
+    PortGroup       qmake 1.0
+
+} else {
+    PortGroup       qmake5 1.0
+
+    qt5.depends_component   qttools qtwebkit-examples
```
Comment (kencu) [inline www/smtube/Portfile:11]:
```suggestion
    qt5.depends_component   qttools qtwebkit
```

Comment (kencu) [reply www/smtube/Portfile:11]: It seems unlikely that a build would depend on qtwebkit-examples rather than qtwebkit... what do you think?

Comment (barracuda156) [reply www/smtube/Portfile:11]: @kencu I believe, you are right.

I just saw the source mentions `Qt5WebKitWidgets`:
https://github.com/smplayer-dev/smtube/blob/d44edf344684ab9bb4117dd031e926b042989cd1/smtube.spec#L31
And only found this in variables: https://github.com/macports/macports-ports/blob/de80cfa1d564f5986397f76eb25ca865e1a87bd7/_resources/port1.0/group/qt5_variables-1.0.tcl#L358-L363
But it is indeed unlikely that examples will be needed. Let me try changing that.

**Comment by kencu** [comment]
@lastic has not been around for some years now. He was very keen and helpful at the beginning, thought. If you wish to be maintainer, that would be fine.

**Comment by barracuda156** [comment]
@kencu Thank you for merging!

**Pull request merged**